test/gurka/test_avoids.cc

```diff
@@ -214,6 +217,50 @@ TEST_P(AvoidTest, TestAvoid2Polygons) {
   };
 }
 
+TEST_P(AvoidTest, TestAvoidShortcuts) {
+  auto node_b = avoid_map.nodes.at("B");
+  auto node_c = avoid_map.nodes.at("C");
+  auto node_e = avoid_map.nodes.at("E");
+  auto dx = std::abs(node_c.lng() - node_b.lng());
+  auto dy = std::abs(node_b.lat() - node_e.lat());
+
+  // create a small polygon on the shortcut: should fail if shortcuts are also avoided
+  //                x-x
+  //      A------B--|-|-C--F--G
+  //                x-x
+  //             |_________|
+  //               shortcut
+
+  // one clockwise ring
+  std::vector<ring_bg_t> rings;
+  rings.push_back({{node_b.lng() + 0.1 * dx, node_b.lat() + 0.01 * dy},
+                   {node_c.lng() - 0.1 * dx, node_c.lat() + 0.01 * dy},
+                   {node_c.lng() - 0.1 * dx, node_c.lat() - 0.01 * dy},
+                   {node_b.lng() + 0.1 * dx, node_b.lat() - 0.01 * dy},
+                   {node_b.lng() + 0.1 * dx, node_b.lat() + 0.01 * dy}});
+
+  // build request manually for now
+  auto lls = {avoid_map.nodes["A"], avoid_map.nodes["G"]};
+
+  // TODO: remove before commit
+  std::cout << gurka::dump_geojson_graph(avoid_map) << std::endl;
```

Comment:
so, this test should print "it worked" right now but doesn't. Actually it's not a good test tbh, it wouldn't really fail if there were no shortcuts in avoid edges set.

Anyways, the intent is that we check that shortcuts are being avoided as well. Somehow the request from "A" to "G" doesn't use the shortcut (BCF) apparently. Anyone sees the problem here with the map or request?

Reply:
crap i should have said something earlier. we only start using shortcuts once the route starts doing hierarchy cully that will never happen with short routes like this. i have another idea for a test though, much simpler!

simply call the `do_action` api in gurka but hold on to the API object that it returns and inspect that for your test. all you have to do is make sure that the edge id for the shortcut is in the list of avoid polygon edges. no need to check that the routing avoided it, the regression we found was that the edge wasnt making it into the set of edges to exclude in the first place not whether the routing algorithm was actually excluding them or not. does that make sense?

Reply:
makes a lot more sense than what I was trying to do :smile:

Reply:
I had to do an actual unit test here. since the route is supposed to fail there's no API object to inspect.